modules/imgcodecs/test/test_png.cpp

```diff
@@ -186,6 +186,32 @@ const string exif_files[] =
 INSTANTIATE_TEST_CASE_P(ExifFiles, Imgcodecs_PNG_Exif,
     testing::ValuesIn(exif_files));
 
+
+TEST(Imgcodecs_Png, pngsuite_decode)
+{
+    const string root = cvtest::TS::ptr()->get_data_path();
+    const string fs_filename = root + "pngsuite/pngsuit_unchanged.xml";
+    const string folder = root + "pngsuite/*.png";
+    vector<String> filenames;
+    glob(folder, filenames);
+
+    FileStorage fs(fs_filename, FileStorage::READ);
+    for (size_t i = 0; i < filenames.size(); i++)
+    {
```

Comment:
It is better to have single file per test case (you don't want to debug issue of 15th image of the test).

Reply:
@alalek, I already investigated this issue. Since it's many tiny files, one big file is probably preferable for now. I suggest to merge as-is

Reply:
This is a general of for writing tests. All-in-one test is anti-pattern.

They should be spitted:
- again, as I mentioned - we need possibility to debug exact case without skipping other 10 cases (debugging code is time consuming task, no need to make it harder)
- test code may do nothing if `glob()` returns nothing due to misconfiguration - and we will not see this in test logs
- all-in-one approach blocks test filtering, disabling